src/server/common/controller/summary-page-controller/SummaryPageController.js

```diff
@@ -0,0 +1,84 @@
+/** @import SummaryPage from '../../model/page/summary-page/SummaryPageModel.js' */
+/** @import { SectionModel } from "~/src/server/common/model/section/section-model/index.js" */
+/** @import { Server, ServerRegisterPluginObject } from '@hapi/hapi' */
+
+export class SummaryPageController {
+  /** @type {string} */
+  indexView = 'common/controller/summary-page-controller/index.njk'
+
+  /** @type {string} */
+  pageTitle
+
+  /** @type {string} */
+  heading
+
+  /**
+   * @param {SummaryPage} page
+   */
+  constructor(page) {
+    this.page = page
+  }
+
+  get urlPath() {
+    return `/${this.page.sectionKey}/summary`
+  }
+
+  /** @type {SectionModel} */
+  Model
+
+  /** @returns {ServerRegisterPluginObject<void>} */
+  plugin() {
+    return {
+      plugin: {
+        name: `${this.page.sectionKey}-summary`,
+
+        /** @param {Server} server */
+        register: (server) => {
+          server.route([
+            {
+              method: 'GET',
+              path: this.urlPath,
+              handler: this.getHandler.bind(this)
+            },
+            {
+              method: 'POST',
+              path: this.urlPath,
+              handler: this.postHandler.bind(this)
+            }
+          ])
+        }
+      }
+    }
+  }
+
+  getHandler(req, res) {
+    const section = this.page.sectionFactory(req.yar.get('origin'))
+
+    const { isValid, firstInvalidPage } = section.validate()
+    if (!isValid) {
+      return res.redirect(
+        `${firstInvalidPage?.urlPath}?redirect_uri=/origin/summary`
```

Comment:
we can't always redirect to origin/summary as this will be used in the other sections